account_asset_management/report/account_asset_report_xls.py

```diff
@@ -23,11 +22,6 @@ class AssetReportXlsx(models.AbstractModel):
     _description = "Dynamic XLS asset report generator"
     _inherit = "report.report_xlsx.abstract"
 
-    def _(self, src):
```

Comment:
Why removing this?

Reply:
The purpose of this method is to get the translation of self.().

The def translate method does not exist in the tools.

Also, in v15 when you run the "def translate" method you also it run the query. 

	`cr.execute('select value from ir_translation where lang=%s and type=%s and 	name=%s and src=%s and md5(src)=md5(%s)', (lang, source_type, str(name), 	source, source))`

But if it is called like this:

	 `val = translate(self.env.cr, IR_TRANSLATION_NAME, "report", lang, src) or src`

The value of type is report
	
        `”type=”report”`

But type in the class ir_translation only can have this values: 
	```
TRANSLATION_TYPE = [
	('model', 'Model Field'),
	('model_terms', 'Structured Model Field'),
	('code', 'Code'),
	]
        ```

If the query does not find anything, it will always return src. 

Also in v15 is not working.

![Selección_063](https://user-images.githubusercontent.com/6359121/196727359-9fbe2d82-2f16-4e6c-b265-7832bb27ffb0.jpg)

I change it to standard Odoo and now is working.

![Selección_064](https://user-images.githubusercontent.com/6359121/196727686-88e62e0a-43ff-47f0-9d66-8ee9aa866d89.jpg)

Reply:
OK, the idea behind this is that the translation works having a context in the class. This trick was needed for v13 if I don't remember bad, but people doing later translations didn't check...